File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.kt
fun solution(n: Int, a: IntArray): Int {
    a.sort()

    var left = 0
    var maxKeep = 0

    for (right in 0 until n) {
        while (a[right] - a[left] >= n) {
            left++
        }

        val currentKeep = right - left + 1
        if (currentKeep > maxKeep) {
            maxKeep = currentKeep
        }
    }

    val minMoves = n - maxKeep
    return minMoves
}

File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.py
def solution(n, a):
    a.sort()
    left = 0
    max_keep = 0

    for right in range(n):
        while a[right] - a[left] >= n:
            left += 1

        current_keep = right - left + 1
        if current_keep > max_keep:
            max_keep = current_keep

    min_moves = n - max_keep
    return min_moves


